Translate the component using locales and translations

Luisetelo opened the pull request:
Hey there! 

I created this PR from my crazy idea of #127. In fact, it wasn't as crazy as I thought. Actually, you can create a picker passing a moment object as `default` paramater. All I did was using this moment object in the points where a translation is needed. As well, I included two new parameters in the constructor to translate the button labels.

jwhitfieldseed commented:
Thanks for the new PR! I'm having a very busy week, but I'll review it when I get the chance. ⭐️

joews commented on lib/js/index.js:
```diff
@@ -40,7 +40,9 @@ const defaults = () => ({
-  dateValidator: undefined
+  dateValidator: undefined,
+  okLabel: 'OK',
+  cancelLabel: 'Cancel'
```
👍 

Can you add these options to README.md?

Luisetelo replied:
Sure!

joews commented on lib/js/index.js:
```diff
@@ -91,10 +95,13 @@ class DateTimePicker extends Events {
       this.setDate(this.options.default);
       this.setTime(this.options.default);
+    } else {
+      this.setDate(this.value);
```
Thanks for this fix 👍 

I applied this part of the patch straight to master to fix the issue. It was released in  [v2.4.0](https://github.com/ripjar/material-datetime-picker/releases/tag/v2.4.0), where I credited you.

Can you remove it from this PR?

Luisetelo replied:
Consider it done :)

joews commented on lib/js/index.js:
```diff
@@ -202,7 +209,7 @@ class DateTimePicker extends Events {
 
   onChangeDate(dateString) {
-    const newValue = moment(this.value);
+    const newValue = moment(this.options.default);
```
This doesn't look right. Can you explain it?

Luisetelo replied:
I did it because when you clicked on a different date it changes the language back to English. This is caused by using `this.value`, which is the default moment object that has the English locale only available. `this.options.default` is the received moment object and can containts, therefore, more languages.

However, as you said it doesnt look right. Looking deeper on the code I have found other cases where this unwelcome behaviour appear.

joews commented on lib/js/index.js:
```diff
@@ -342,7 +349,7 @@ class DateTimePicker extends Events {
     const m = moment(date);
     const month = m.format('MMM');
-    const day = m.format('Do');
+    const day = m.format('D');
```
Whilst this change better matches the material "picker" examples, I think it makes the component feel visually unbalanced.

Can you add options for the number strings instead?

Luisetelo replied:
Let me check if there is an option in moment that I can use depending on the language

joews commented on lib/js/index.js:
```diff
@@ -70,15 +72,17 @@ class DateTimePicker extends Events {
-      initialValue: this.value
+      initialValue: this.options.default,
+      weekdayFormat: this.options.default.localeData().weekdaysMin(),
+      weekStart: this.options.default.clone().weekday(0).day()
```
What do these two lines change?

Luisetelo replied:
`WeekDayFormat` changes the names of the days displayed on the calendar according to the language.
`weekStart` sets the week start day according to the moment object received as parameter.

[Source](https://github.com/bevacqua/rome)

joews reviewed (changes requested):
Thanks for the change! I'm sorry it has taken a while to review. I have a few questions before I'm ready to merge it.

falko100 commented:
I want this ! 👍

arist0tl3 commented:
Just wanted to say thanks for this @Luisetelo!

Luisetelo committed:
Translation feature

Luisetelo commented:
Working on an improved version...!

Luisetelo committed:
Merge branch 'master' into translate

Luisetelo commented:
Hey there!

Here is the new version of the feature. The changes I've made are:

- Added translations for Ok and Cancel buttons
- Added translations for the calendar (day names, first day of the week)
- Added day number format
- Added documentation of new parameters at README.md
- Updated moment.js dependency (see below)
- Deleted compiled files for the PR
- Deleted blank spaces in lines

**Drawback**
In order to use this new feature, moment.js dependency **should be updated** . Why? To set the day names translations in the calendar, the function `moment.weekdaysMin()` is used. This function list the day names from the current locale. However, **this behavior is only available in moment.js 2.13.0 and above** ([see](https://momentjs.com/docs/#/i18n/listing-months-weekdays/))

Looking forward to hearing from you!
